lib/ansible/plugins/action/iosxr.py

```diff
@@ -62,5 +65,18 @@ def run(self, tmp=None, task_vars=None):
 
             task_vars['ansible_socket'] = socket_path
 
+        # make sure we are in the right cli context which should be
+        # enable mode and not config module
+        if socket_path is None:
+            socket_path = self._connection.socket_path
+
+        conn = Connection(socket_path)
+        out = conn.get_prompt()
+        while to_text(out, errors='surrogate_then_replace').strip().endswith(')#'):
+            display.vvvv('wrong context, sending exit to device', self._play_context.remote_addr)
+            conn.send_command('exit')
```

Comment:
If there are uncommitted changes on the prompt, this will not exit.
Assuming there are no uncommitted changes, you can use `end` instead of `exit` to exit from configure mode.

If stale uncommitted changes are expected by any chance; use`abort`. This will abort any uncommitted changes and bring prompt out of configure mode. 

This may apply to other platforms as well.

Reply:
+1 to `end`